idaes_examples/notebooks/docs/unit_models/custom_unit_models/liquid_extraction.py

```diff
@@ -0,0 +1,596 @@
+        ),
+    )
+    CONFIG.declare(
+        "liquid_property_package_args",
+        ConfigBlock(
+            implicit=True,
+            description="Arguments to use for constructing liquid phase properties",
+            doc="""A ConfigBlock with arguments to be passed to liquid phase
+property block(s) and used when constructing these,
+**default** - None.
+**Valid values:** {
+see property package for documentation.}""",
+        ),
+    )
+    CONFIG.declare(
+        "aqueous_property_package",
+        ConfigValue(
+            default=useDefault,
+            domain=is_physical_parameter_block,
+            description="Property package to use for aqueous phase",
+            doc="""Property parameter object used to define property calculations
+for the aqueous phase,
+**default** - useDefault.
+**Valid values:** {
+**useDefault** - use default package from parent model or flowsheet,
+**PropertyParameterObject** - a PropertyParameterBlock object.}""",
+        ),
+    )
+    CONFIG.declare(
+        "aqueous_property_package_args",
+        ConfigBlock(
+            implicit=True,
+            description="Arguments to use for constructing aqueous phase properties",
+            doc="""A ConfigBlock with arguments to be passed to aqueous phase
+property block(s) and used when constructing these,
+**default** - None.
+**Valid values:** {
+see property package for documentation.}""",
+        ),
+    )
+    CONFIG.declare(
+        "reaction_package",
+        ConfigValue(
+            default=None,
+            domain=is_reaction_parameter_block,
+            description="Reaction package to use for control volume",
+            doc="""Reaction parameter object used to define reaction calculations,
+**default** - None.
+**Valid values:** {
+**None** - no reaction package,
+**ReactionParameterBlock** - a ReactionParameterBlock object.}""",
+        ),
+    )
+    CONFIG.declare(
+        "reaction_package_args",
+        ConfigBlock(
+            implicit=True,
+            description="Arguments to use for constructing reaction packages",
+            doc="""A ConfigBlock with arguments to be passed to a reaction block(s)
+and used when constructing these,
+**default** - None.
+**Valid values:** {
+see reaction package for documentation.}""",
+        ),
+    )
+
+    def build(self):
+        """
+        Begin building model (pre-DAE transformation).
+        Args:
+            None
+        Returns:
+            None
+        """
+        # Call UnitModel.build to setup dynamics
+        super(LiqExtractionData, self).build()
+
+        # Check phase lists match assumptions
+        if self.config.aqueous_property_package.phase_list != ["Aq"]:
+            raise ConfigurationError(
+                f"{self.name} Liquid-Liquid Extractor model requires that the aquoues "
+                f"phase property package have a single phase named 'Aq'"
+            )
+        if self.config.liquid_property_package.phase_list != ["Liq"]:
```

Comment:
This naming is a bit odd - an aqueous phase is by definition a liquid phase. I would recommend you change this to organic. Also, I suggest making the aqueous phase `aqueous` for clarity.

Finally, do you actually need to be this restrictive? Is it not sufficient to just check that each property package supports a single phase? (I.e. let the name be arbitary).